src/Common/Config/ConfigReloader.cpp

```diff
@@ -131,18 +136,22 @@ std::optional<ConfigProcessor::LoadedConfig> ConfigReloader::reloadIfNewer(bool
             if (Coordination::isHardwareError(e.code))
                 need_reload_from_zk = true;
 
+            PreformattedMessage message = getCurrentExceptionMessageAndPattern(true);
+            message.text = fmt::format("ZooKeeper error when loading config from '{}' config: {}", config_path, message.text);
             if (throw_on_error)
-                throw;
+                throw Exception(ErrorCodes::CANNOT_RELOAD_CONFIG, message);
 
-            tryLogCurrentException(log, "ZooKeeper error when loading config from '" + config_path + "'");
+            LOG_ERROR(log, message)
             return std::nullopt;
         }
         catch (...)
         {
+            PreformattedMessage message = getCurrentExceptionMessageAndPattern(true);
+            message.text = fmt::format("Error loading config from '{}' config: {}", config_path, message.text);
             if (throw_on_error)
-                throw;
+                throw Exception(ErrorCodes::CANNOT_RELOAD_CONFIG, message);
 
-            tryLogCurrentException(log, "Error loading config from '" + config_path + "'");
+            LOG_ERROR(log, message)
```

Comment:
```suggestion
            LOG_ERROR(log, "Error loading config from '{}' {} {}", config_path, e.what(), StackTrace().toString());
```

I think you may change all the `LOG_ERROR` places to it